src/main/java/com/minecraftabnormals/abnormals_core/core/annotations/ConfigKey.java

```diff
@@ -0,0 +1,50 @@
+package com.minecraftabnormals.abnormals_core.core.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation, when applied to {@link net.minecraftforge.common.ForgeConfigSpec.ConfigValue ConfigValue} fields
+ * in objects that are passed into the {@code Object...} parameter of
+ * {@link com.minecraftabnormals.abnormals_core.core.util.DataUtil#registerConfigCondition(String, Object...) registerConfigCondition},
+ * allows the value to be referred to in JSON by a
+ * {@link com.minecraftabnormals.abnormals_core.core.api.conditions.ConfigValueCondition ConfigValueCondition} matching
+ * the mod ID passed into {@link com.minecraftabnormals.abnormals_core.core.util.DataUtil#registerConfigCondition(String, Object...) registerConfigCondition},
+ * which can then be used to test a condition.
```

Comment:
I think some of these docs here can be condensed on the same line.

Reply:
That's fair enough, cause it looks fine when the docs are actually formatted- it's just that the links are annoyingly long. I split the lines to try and make it more readable unformatted, but i guess it doesn't matter much since most ides let you view processed javadocs in some form.